### ISSUE TITLE
[16.0] fs_storage + EDI: connection closed after some time

### ISSUE THREAD

**Issue by JordiMForgeFlow** [OCA/storage#319]
## Module
fs_storage

## Describe the bug
I am using fs_storage to connect to SFTP with EDI (the PR is still not merged https://github.com/OCA/edi-framework/pull/26). The connection works fine, but after some time the EDI jobs start to fail due to the following issue:

```
File "/opt/odoo/auto/addons/fs_storage/models/fs_storage.py", line 376, in add
    self.fs.makedirs(self.fs.sep.join(path))
  File "/usr/local/lib/python3.10/site-packages/fsspec/implementations/sftp.py", line 89, in makedirs
    self.ftp.mkdir(new_path, mode)
  File "/usr/local/lib/python3.10/site-packages/paramiko/sftp_client.py", line 460, in mkdir
    self._request(CMD_MKDIR, path, attr)
  File "/usr/local/lib/python3.10/site-packages/paramiko/sftp_client.py", line 856, in _request
    num = self._async_request(type(None), t, *args)
  File "/usr/local/lib/python3.10/site-packages/paramiko/sftp_client.py", line 881, in _async_request
    self._send_packet(t, msg)
  File "/usr/local/lib/python3.10/site-packages/paramiko/sftp.py", line 209, in _send_packet
    self._write_all(out)
  File "/usr/local/lib/python3.10/site-packages/paramiko/sftp.py", line 173, in _write_all
    n = self.sock.send(out)
  File "/usr/local/lib/python3.10/site-packages/paramiko/channel.py", line 799, in send
    return self._send(s, m)
  File "/usr/local/lib/python3.10/site-packages/paramiko/channel.py", line 1196, in _send
    raise socket.error("Socket is closed")
OSError: Socket is closed
```
If I press the button 'TEST CONNECTION' in the fs storage, I also get the same error. However, if I restart Odoo, it starts working fine again.

My hint is that the problem comes from fs_storage storing the connection in a variable (https://github.com/OCA/storage/blob/3083166c5fbf4e4a888babfc916caf9f6f530034/fs_storage/models/fs_storage.py#L274). The connection closes at some point (probably due to timeout) and it does not work anymore.

@lmignon @sebastienbeau have you faced something like this with fs_storage?

**Comment by lmignon** [comment]
@JordiMForgeFlow I never encountered this kind of error. You could patch the code to not store the filesystem objet on a variable to see if it solve the problem...

**Comment by JordiMForgeFlow** [comment]
@lmignon thanks for the feedback. I will try and let you know the outcome

**Comment by nilshamerlinck** [comment]
Hello @JordiMForgeFlow I faced a similar issue where the underlying ssh connection seems to be prematurely closed. Adding a timeout helped as a workaround. Can you try adding `{ "timeout": 2 }`in your `fs.storage` `options`?

**Comment by JordiMForgeFlow** [comment]
> Hello @JordiMForgeFlow I faced a similar issue where the underlying ssh connection seems to be prematurely closed. Adding a timeout helped as a workaround. Can you try adding `{ "timeout": 2 }`in your `fs.storage` `options`?

Thank you for the hint @nilshamerlinck, I will give it a try and see if that solves it

**Comment by JordiMForgeFlow** [comment]
@nilshamerlinck @lmignon After trying different things I have managed to get it working with the following changes: https://github.com/OCA/storage/pull/320

I have suggested them as settings in the FS Storage.

Apart from storing the connection, I also saw that fsspec instances also cache information. Invalidating the cache when the connection gives an error allowed to only force the reconnection when needed.
